runtime/compiler/control/J9Options.cpp

```diff
@@ -1735,13 +1735,14 @@ J9::Options::fePreProcess(void * base)
 
          UDATA largePageSize = 0;
          UDATA largePageFlags = 0;
-         int32_t lpArgIndex;
          UDATA lpSize = 0;
+         int32_t lpArgIndex = FIND_ARG_IN_VMARGS(EXACT_MEMORY_MATCH, "-Xlp", 0);
 
          // -Xlp<size>, attempt to use specified page size
-         if ((lpArgIndex=FIND_ARG_IN_VMARGS(EXACT_MEMORY_MATCH, "-Xlp", 0)) >= 0)
+         if (lpArgIndex >= 0)
             {
-            GET_MEMORY_VALUE(lpArgIndex, "-Xlp", lpSize);
+            char *lpOption = "-Xlp";
```

Comment:
Is this assignment a compiler workaround? I'm wondering why the `"-Xlp"` isn't hard coded in the call to GET_MEMORY_VALUE. Really it should be a #define, like those found in jvminit.h

Reply:
Hard coding -Xlp to GET_MEMORY_VALUE does cause the program to crash. So It is some sort of workaround. Taking a closer look at it to see if it's a simple fix.

Reply:
If a workaround is needed, please add a comment to explain the code.

Reply:
@pshipton It looks like it's a undocumented workaround. In all use cases of GET_MEMORY_VALUE, we've always passed in the option as an argument rather than directly specifying the string as an argument. 

The expected behaviour is that they should behave similarly. 

I suspect it's another compiler bug. Shall we create another issue to track this?

Reply:
Not sure of the value of an Issue in this case, but you can open one if you want. I'd like to see a comment in the code here to explain why lpOption is used. This can avoid somebody rewriting the code and introducing a problem.

Reply:
In a more thorough review it doesn't look like a bug. 

The macro conversion done by GET_MEMORY_VALUE  is (char **)&lpOption. It should be expected that there should be different behaviours when passing a pointer to string rather than referencing the string itself. 

As you've suggested I'm going to add a comment explaining why it was used.

Reply:
See 5188840ef2149d430ab6579f86b25e71cb8c27ad